compilation-support-image/compile-rust-wasi.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+while IFS= read -r line || [ -n "$line" ]
+do
+  echo "$line" >> tmp.rs
+done < "/dev/stdin"
+
+rustc tmp.rs --target=wasm32-wasi -C opt-level=0 -C debuginfo=0 -o tmp.wasm > /dev/null 2> /dev/null
```

Comment:
I couldn't actually get this to work on my machine. The docker image is x86, so i made my own. but for some reason it's failing to create these tmp files. It gets to this point and then says that there is no file. It must be failing to run line 5.

Reply:
That's wired, thanks for reporting!
Unfortunately, I don't have an ARM machine to test, so I'll need someone's help to debug this one.

Reply:
well, docker images are ISA specific. But i may be able to detect if this is running in mac and force it to emulate x86. Will try some things out don't worry about it for now.